app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/adapters/HomeListAdapter.kt

```diff
@@ -0,0 +1,62 @@
+package au.com.gridstone.trainingkotlin.adapters
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.FrameLayout
+import android.widget.ImageView
+import android.widget.TextView
+import androidx.annotation.NonNull
+import androidx.recyclerview.widget.RecyclerView
+import au.com.gridstone.trainingkotlin.Constants.IMAGE_URL
+import au.com.gridstone.trainingkotlin.R
+import au.com.gridstone.trainingkotlin.data.PokemonResults
+import au.com.gridstone.trainingkotlin.data.Pokemon
+import au.com.gridstone.trainingkotlin.views.DetailViewController
+import com.bluelinelabs.conductor.Router
+import com.bluelinelabs.conductor.RouterTransaction
+import com.bluelinelabs.conductor.changehandler.VerticalChangeHandler
+import com.squareup.picasso.Picasso
+
+internal class HomeListAdapter(private var itemsList: List<Pokemon>, private val router: Router) :
+  RecyclerView.Adapter<HomeListAdapter.MyViewHolder>() {
+  internal inner class MyViewHolder(view: View) : RecyclerView.ViewHolder(view) {
+    var itemView: FrameLayout = view.findViewById(R.id.view)
+    var image: ImageView = view.findViewById(R.id.image_list)
+    var name: TextView = view.findViewById(R.id.name)
+  }
+
+  @NonNull
+  override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
+    val itemView = LayoutInflater.from(parent.context)
+      .inflate(R.layout.item_home_recyclerview, parent, false)
+    return MyViewHolder(itemView)
+  }
+
+  override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
+    val item = itemsList[position]
+    holder.name.text = item.name
+    loadImage(holder, item)
+
+    holder.itemView.setOnClickListener {
+      TODO("Working on this next")
+    }
+  }
+
+  private fun loadImage(holder: MyViewHolder, item: Pokemon) {
+    var imageNumbers = item.url.split("/").dropLast(1).last()
+    Picasso.get()
+      .load("$IMAGE_URL/$imageNumbers.png")
+      .placeholder(R.drawable.ic_placeholder)
+      .into(holder.image)
+  }
+
+  override fun getItemCount(): Int {
+    return itemsList.size
+  }
+
+  fun setList(it: PokemonResults?) {
```

Comment:
why is this taking in a nullable value? 🤔

Reply:
it was before when I had sent data without implementing the sealed classes artitechure. no by going through the code and by dry running, I realize that it won't need this nullable anymore thanks for mentioning I had removed this.